array_api_strict/_array_object.py

```diff
@@ -193,6 +215,15 @@ def _check_allowed_dtypes(self, other: bool | int | float | Array, dtype_categor
 
         return other
 
+    def _check_device(self, other):
+        """Check that other is on a device compatible with the current array"""
+        if isinstance(other, (int, complex, float, bool)):
+            return other
+        elif isinstance(other, Array):
+            if self.device != other.device:
+                raise RuntimeError(f"Arrays from two different devices ({self.device} and {other.device}) can not be combined.")
+            return other
```

Comment:
I don't see a need for this function to return something. `_check_allowed_dtypes` returns something because it can return `NotImplemented`, but here the return object is always `other`.

Reply:
Fine for me. I had it return the input just because it would match the API of the other check. Changed now.